src/Parlot/Character.cs

```diff
@@ -68,75 +68,69 @@ public static char ScanHexEscape(string text, int index, out int length)
             return (char)code;
         }
 
-        public static TextSpan DecodeString(string s) => DecodeString(new TextSpan(s));
+        public static BufferSpan<char> DecodeString(string s) => DecodeString(s.ToCharArray());
```

Comment:
```suggestion
        public static BufferSpan<char> DecodeString(string s) => DecodeString(new BufferSpan<char>(s));
```

Reply:
I thought so initially, but eventually, you need to build a char[] as you might be removing some characters if there are any escape to happen